packages/react/src/components/Table/TableHeader.tsx

```diff
@@ -1,21 +1,72 @@
-import React, { ThHTMLAttributes } from 'react';
+import React, { useRef, useEffect, ThHTMLAttributes } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import Icon from '../Icon';
+
+// these match aria-sort's values
+type SortDirection = 'ascending' | 'descending' | 'none';
 
 interface TableHeaderProps extends ThHTMLAttributes<HTMLTableCellElement> {
   children: React.ReactNode;
+  sortDirection?: SortDirection;
+  onSort?: () => void;
   className?: string;
 }
 
-const TableHeader = ({ children, className, ...other }: TableHeaderProps) => (
-  <th className={classNames('TableHeader', className)} {...other}>
-    {children}
-  </th>
-);
+const TableHeader = ({
+  children,
+  sortDirection,
+  onSort,
+  className,
+  ...other
+}: TableHeaderProps) => {
+  const sortButtonRef = useRef<HTMLButtonElement>(null);
```

Comment:
I think if all of the re-rendering bugs are fixed, then you shouldn't have to manage focus manually. The reason why the focus was being lost was because the button was being removed and added back to the DOM.

As an aside, right now, I can get into this state by clicking on a sortable header and then clicking on a non-sortable header:
![image](https://user-images.githubusercontent.com/6416791/150187072-ee247963-2244-4007-b17f-8d83ce94295c.png)

As an aside aside, is that the highlight color Aaron requested? It seems odd that the highlight color would be the same color as the background.

Reply:
Good call for the focus! And yes, I double checked. That's the color for the highlight.
<img width="380" alt="Screen Shot 2022-01-19 at 2 59 18 PM" src="https://user-images.githubusercontent.com/83524297/150204712-c0833be8-c3dd-4c34-aa23-e86e21a0453d.png">